src/NHapi.Base/Util/SynchronizedCache.cs

```diff
@@ -0,0 +1,144 @@
+﻿#if NETFRAMEWORK
```

Comment:
Should this be `NET35` instead of `NETFRAMEWORK`?

Reply:
I thought I might not be able to unit test it since the unit tests target later versions of .NET Framework but it seems to work okay.

Made the change.